docs/docs/smart-contracts/PSP1155/psp1155.md

```diff
@@ -0,0 +1,149 @@
+
+[features]
+default = ["std"]
+std = [
+   "ink_primitives/std",
+   "ink_metadata",
+   "ink_metadata/std",
+   "ink_env/std",
+   "ink_storage/std",
+   "ink_lang/std",
+   "scale/std",
+   "scale-info",
+   "scale-info/std",
+
+   # These dependencies   
+   "psp1155/std",
+   "brush/std",
+]
+```
+
+## Step 2: Add imports
+
+Replace `ink::contract` macro by `brush::contract`.
+Import **everything** from `psp1155::traits`.
+
+```rust
+#[brush::contract]
+pub mod my_psp721 {
+   use psp1155::traits::*;
+```
+
+## Step 3: Define storage
+
+Declare storage struct and declare the fields related to `PSP1155Storage` and `PSP1155MetadataStorage`
+traits. Then you need to derive `PSP1155Storage` and `PSP1155MetadataStorage` traits and mark corresponding fields
+with `#[PSP1155StorageField]` and `#[PSP1155MetadataStorageField]` attributes. Deriving these traits allows you to reuse
+the default implementation of `PSP1155` and `PSP1155Metadata`.
+
+```rust
+#[derive(Default, PSP1155Storage, PSP1155MetadataStorage)]
+#[ink(storage)]
+pub struct MyPSP1155 {
+    #[PSP1155StorageField]
+    psp1155: PSP1155Data,
+    #[PSP1155MetadataStorageField]
+    metadata: PSP1155MetadataData,
+}
+```
+
+## Step 4: Inherit logic
+
+Inherit implementations of `PSP1155` and `PSP1155Metadata` traits. You can customize (override) methods in this `impl` block.
+
+```rust
+impl PSP1155 for MyPSP1155 {}
+
+impl PSP1155Metadata for MyPSP1155 {}
+```
+
+## Step 5: Define constructor
+
+Define constructor. Your basic version of `PSP1155` contract is ready!
+
+```rust
+impl MyPSP1155 {
+    #[ink(constructor)]
+    pub fn new(name: Option<String>, symbol: Option<String>) -> Self {
+        let mut instance = Self::default();
+        instance.metadata.name = name;
+        instance.metadata.symbol = symbol;
+        instance
+    }
+}
+```
+
+## Step 6: Customize your contract
+
+Customize it by adding logic for adding new token types which can be minted. We will raise the value of `unique_ids` each time we add new token type. New token types will also be added to `registered_ids` and when new tokens are about to be minted, we will first check if the type of minted token is registered in the smart contract.
+
+```rust
+#[derive(Default, PSP1155Storage, PSP1155MetadataStorage)]
+#[ink(storage)]
+pub struct MyPSP1155 {
+    #[PSP1155StorageField]
+    psp1155: PSP1155Data,
+    #[PSP1155MetadataStorageField]
+    metadata: PSP1155MetadataData,
+    unique_ids: u8,
+    registered_ids: StorageHashMap<Id, bool>,
+}
+
+impl PSP1155 for MyPSP1155 {}
+
+impl PSP1155Metadata for MyPSP1155 {}
+
+impl MyPSP1155 {
+    #[ink(constructor)]
+    pub fn new(name: Option<String>, symbol: Option<String>) -> Self {
+        let mut instance = Self::default();
+        instance.metadata.name = name;
+        instance.metadata.symbol = symbol;
+        instance
+    }
+
+    #[ink(message)]
+    pub fn add_type(&mut self) {
```

Comment:
It is not clear, let's change the signature of the method like:
`pub fn add_type(&mut self, allowed_id: [u8; 32])`. It means that the caller can specify which id can be minted=)

Reply:
fixed in the example as well